packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -20,12 +20,60 @@ class MultiChildLayoutParentData extends ContainerBoxParentDataMixin<RenderBox>
 
 /// A delegate that controls the layout of multiple children.
 ///
+/// Delegates must be immutable. Specifically, if two delegates are equal, then
+/// then must produce the same layout. To change the layout, replace the
+/// delegate with a different instance.
```

Comment:
maybe we should document and even assert that x.shouldRelayout(x) returns false?